prime-dotnet-webapi/Services/PartyService.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Prime.Models;
+
+
+namespace Prime.Services
+{
+    public class PartyService : BaseService, IPartyService
+    {
+        public PartyService(
+            ApiDbContext context,
+            IHttpContextAccessor httpContext)
+            : base(context, httpContext)
+        { }
+
+        public async Task<bool> PartyExistsAsync(int partyId)
+        {
+            return await GetBasePartyQuery()
+                .AnyAsync(e => e.Id == partyId);
+        }
+
+        public async Task<bool> PartyUserIdExistsAsync(Guid userId)
+        {
+            return await GetBasePartyQuery()
+                .AnyAsync(e => e.UserId == userId);
+        }
+
+
+        public async Task<Party> GetPartyForUserIdAsync(Guid userId)
+        {
+            return await GetBasePartyQuery()
+                .AsNoTracking()
+                .FirstOrDefaultAsync(e => e.UserId == userId);
+        }
+
+        public async Task<int> CreatePartyAsync(Party party)
+        {
+            if (party == null)
+            {
+                throw new ArgumentNullException(nameof(party), "Could not create an party, the passed in Party cannot be null.");
+            }
+
+            _context.Parties.Add(party);
+
+            var created = await _context.SaveChangesAsync();
+            if (created < 1)
+            {
+                throw new InvalidOperationException("Could not create party.");
+            }
+
+            return party.Id;
+        }
+
+        public async Task<int> UpdatePartyAsync(int partyId, Party party)
+        {
+            var currentParty = await GetBasePartyQuery()
+                .SingleAsync(e => e.Id == partyId);
+
+            _context.Entry(party).CurrentValues.SetValues(party);
+
+            UpdatePartyAddress(currentParty, party);
+
+            try
+            {
+                return await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return 0;
+            }
+        }
+
+        public void UpdatePartyAddress(Party current, Party updated)
```

Comment:
Dont think this method handles the case where the new party's address is null

Reply:
Yeah it does this adds it if null otherwise it updates it
` if (current.PhysicalAddress == null)
  {
        current.PhysicalAddress = updated.PhysicalAddress;
   }`